### ISSUE TITLE
"compact(...) can be used" triggered when destructuring an associative array

### ISSUE THREAD

**Issue by hackel** [kalessil/phpinspectionsea#1658]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) 4.0.6 |
| Language level | PHP 7.4                         |

Current behaviour
--
```php
$location = ['city' => 'Paris', 'country' => 'France'];

['city' => $city, 'country' => $country] = $location;
^~~~ [EA] 'compact('city', 'country')' can be used instead (improves maintainability).

$label = "The location is $city, $country.";
```

The autofix changes the code to `compact('city', 'country') = $address;` which produces a fatal error (Can't use function return value in write context).

Expected behaviour
--
Inspection should not be triggered when using shorthand array syntax to destructure an array.

Environment details
---
IntelliJ IDEA 2020.2.3 (Ultimate Edition)
Build #IU-202.7660.26, built on October 6, 2020

**Comment by ea-inspections-team** [comment]
Thanks for reporting @hackel! Fixed!
